JetMETCorrections/Modules/plugins/METCorrectorDBWriter.cc

```diff
@@ -24,49 +27,53 @@ class  METCorrectorDBWriter : public edm::EDAnalyzer
  private:
   std::string era;
   std::string algo;
+  std::string path;
   std::string inputTxtFile;
   std::string payloadTag;
 };
+}
 
 METCorrectorDBWriter::METCorrectorDBWriter(const edm::ParameterSet& pSet)
 {
   era    = pSet.getUntrackedParameter<std::string>("era");
   algo   = pSet.getUntrackedParameter<std::string>("algo");
+  path   = pSet.getUntrackedParameter<std::string>("path");
   payloadTag = algo;
 }
 
 void METCorrectorDBWriter::beginJob()
 {
-  std::string path("CondFormats/JetMETObjects/data/");
+  LogDebug ("default")<<"beginJob===========";
 
   METCorrectorParametersCollection *payload = new METCorrectorParametersCollection();
   std::cout << "Starting to import payload " << payloadTag << " from text files." << std::endl;
-  for( int i(0); i< METCorrectorParametersCollection::N_LEVELS;++i)
+  for( int ilev(0); ilev< METCorrectorParametersCollection::N_LEVELS;++ilev)
   {
     std::string append("_");
-    std::string ilev = METCorrectorParametersCollection::findLabel( static_cast<METCorrectorParametersCollection::Level_t>(i) );
-    append += ilev;
+    std::string levelName = METCorrectorParametersCollection::findLabel( static_cast<METCorrectorParametersCollection::Level_t>(ilev) );
+    append += levelName;
     append += "_";
     append += algo;
     append += ".txt";
     inputTxtFile = path+era+append;
-    std::ifstream input( ("../../../"+inputTxtFile).c_str() );
-    if ( input.good() ) {
+    try{
       edm::FileInPath fip(inputTxtFile);
       std::cout << "Opened file " << inputTxtFile << std::endl;
+      // Create the parameter object from file
       std::vector<std::string> sections;
-      METCorrectorParametersCollection::getSections("../../../"+inputTxtFile, sections );
+      METCorrectorParametersCollection::getSections(fip.fullPath(), sections );
       if(sections.size() == 0){
-        payload->push_back(i, METCorrectorParameters(fip.fullPath(),"") );
+        payload->push_back(ilev, METCorrectorParameters(fip.fullPath(),"") );
       }else{
 	for ( std::vector<std::string>::const_iterator isectbegin = sections.begin(), isectend = sections.end(), isect = isectbegin;
 	      isect != isectend; ++isect ) {
-	  payload->push_back( i, METCorrectorParameters(fip.fullPath(),*isect), ilev + "_" + *isect );	  
-	  std::cout << "Added " << ilev + "_" + *isect <<  " to record " << i << std::endl;
+	  payload->push_back( ilev, METCorrectorParameters(fip.fullPath(),*isect), *isect );	  
+	  std::cout << "Added level " << levelName  + "_" + *isect <<  " to record "<<ilev<< std::endl;
 	}
       }
-      std::cout << "Added record " << i << std::endl;
-    }else{
+      std::cout << "Added record " << ilev << std::endl;
```

Comment:
Here `cout` is used, while on line 46 you use `LogDebug`. Is this code used in any workflow, or is it only called by a user?